[CS2113-T12-2] Big-PP

Magmanat opened the pull request:
This is a command-line tool designed to help you pick computer parts based on your needs and budget.

Vignesh-30 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+Below is the Class diagram for the UI class
 
-## Design & implementation
+![UI Class Diagram](uml-pictures/UIClassDiagram.png)
```
Class diagrams should have + and - to denote public and private access. Also, UI class should have abstract denoted as {abstract} instead of using a "A" sign.

de-yi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+## General Overview
+below is the overall architecture diagram for how BigPP works.
+
+![Architecture Diagram](uml-pictures/ArchitectureDiagram.png)
```
Shouldn't the diagrams follow UML standards?

![image](https://user-images.githubusercontent.com/109506859/228425542-523b0601-79cd-48e0-bddc-9df6036fb81d.png)

kairuler replied:
Perhaps if the diagrams were to use these non-standard notation, you guys could put in a legend to tell the readers what each notation mean. It could make it clearer for readers like us!
e.g. what is 'A', 'C', 'E', the red hollow box and green filled circle represent?

de-yi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+
+### PC Class Architecture
+
+![PC Class Diagram](uml-pictures/PcDiagram.png)
```
Shouldn't some details from the diagram can be ommitted?
![image](https://user-images.githubusercontent.com/109506859/228426229-7f52984e-23b4-49d6-8046-4087e5efd360.png)

de-yi replied:
Also, shouldn't it use consistent format?

![image](https://user-images.githubusercontent.com/109506859/228426766-c098de69-77b3-40ec-a076-c2e9f7e2ad83.png)

Vignesh-30 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+
+### PC Class Architecture
+
+![PC Class Diagram](uml-pictures/PcDiagram.png)
```
When viewing this Class diagram on the webpage, it is difficult to read all the components clearly without zooming in.
Maybe, the diagrams could be split into multiple and presented in a different way to make it more readable.

Magmanat replied:
TA said this was ok

Vignesh-30 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
```
Perhaps you could give more user stories to highlight the usages of the application

Magmanat replied:
We are on it!

de-yi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+found below:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![List Component Command](./uml-pictures/listComponentCommand.png)
```
Shouldn't User & BigPP need activation bar? Also, it might be better to "hide foot-box."

![image](https://user-images.githubusercontent.com/109506859/228428504-f65955f2-903e-40db-afe8-9705edf45784.png)

Vignesh-30 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+
+Overall, this class diagram provides a high-level overview of the components that make up a computer system and their relationships to each other.
+
+### Command class
```
Perhaps, you could provide a class diagram for the Command Class to highlight the interactions with other classes

Magmanat replied:
We might do an object diagram for this

Vignesh-30 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+#### `list [COMPONENT]` Command (builder mode)
+The `list [COMPONENT]` command prints out a formatted list of all available components of type `[COMPONENT]`.
+
+When the user inputs a command of the form `list [COMPONENT]` in builder mode,
```
You could also give more information about other types of commands you have in the application

Magmanat replied:
Sounds like a good idea

Manoj364 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
-## Design & implementation
+![UI Class Diagram](uml-pictures/UIClassDiagram.png)
+
+The `UI` Class serves as the interface between the user and BigPP. It is responsible for the following key actions
```
In the UI diagram, UI class is declared as an abstract class, but in the explanation, "interface" is used, so it is slightly confusing if UI is an abstract class or an interface

de-yi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,190 @@
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+
+![load Chassis Sequential Diagram](uml-pictures/loadChassisSequential.png)
```
Shouldn't the activation bar of DataStorage extended?
![image](https://user-images.githubusercontent.com/109506859/228429808-21ea1f8f-ccc2-429c-b1c2-7ea38da565b9.png)

Magmanat committed:
Add functionality for compatibility checks

Magmanat committed:
Update EXPECTED.TXT

Magmanat committed:
Merge pull request #148 from Reynold-SL/branch_developer_guide_sequence

Developer Guide Viewer Add Command sequence diagram

Magmanat committed:
Resolve merge conflicts

Magmanat committed:
Merge pull request #149 from Magmanat/compatibility_check

Compatibility check

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+<!-- omit in toc -->
 # Developer Guide
+<!-- omit in toc -->
```
Not specific to the Developer Guide, but couldn't the navigation bar at the top of the page be adjusted? For example, change the "Duke" link to the name of your own program. A link to the User Guide can also improve the experience.

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+<!-- omit in toc -->
 # Developer Guide
+<!-- omit in toc -->
+## Table of contents
```
Design & implementation appears as `Design \& implementation` in the Table of Contents.

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+![Viewer Add Command Sequential Diagram](uml-pictures/ViewerAddCommand.png)
```
Shouldn't there be an activation bar for ParseViewerCommand?

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+found below:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![List Component Command](./uml-pictures/listComponentCommand.png)
```
Shouldn't BigPP have an activation bar? And since the User isn't an object in the program, should it be in the diagram?

kairuler replied:
There are a few inconsistencies with the activation bars of methods called in the same object. e.g.
![image](https://user-images.githubusercontent.com/78406251/228765247-36962fdb-40de-4613-be8f-a34e158803c3.png)
![image](https://user-images.githubusercontent.com/78406251/228765339-c9e87505-d419-4d0e-a30d-15155367fd5d.png)
I think that the correct way is to use an activation bar for method called in the same object just like in the second photo.

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
-## Non-Functional Requirements
+### Non-Functional Requirements
 
 {Give non-functional requirements}
 
-## Glossary
+### Glossary
 
 * *glossary item* - Definition
 
-## Instructions for manual testing
+### Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
```
Can these sections be removed since they don't contain any information for now?

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
```
Is this a leftover from the template Developer Guide? There is no mention of any to-do list functionality in the rest of the guide.

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+### DataStorage class
+Below is the Class diagram for the DataStorage class:
+
+![DataStorage Class Diagram](uml-pictures/DataStorageClass.png)
```
Do the PATHs in the DataStorage Class need to be included in this diagram? If the PATHs are final, could they be removed to make the diagram more readable?

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+
+![load Chassis Sequential Diagram](uml-pictures/loadChassisSequential.png)
```
Shouldn't the loadAll() method call start at the top of the activation bar and not before it? And could you add activation bars for the self-invoked method calls?

notbingsu replied:
: could be placed before instances instantiated, ie Chassis could be :Chassis

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+![Viewer Add Command Sequential Diagram](uml-pictures/ViewerAddCommand.png)
+### PC Class Architecture
+
+![PC Class Diagram](uml-pictures/PcDiagram.png)
```
Is there supposed to be a gap in GPU and CPUCooler?

kairuler replied:
![image](https://user-images.githubusercontent.com/78406251/228763244-4b388d24-d6d4-43b8-8f66-865fcd5e9e49.png)
![image](https://user-images.githubusercontent.com/78406251/228763309-26ce1d57-4bde-4f53-ab26-766b945d5e8f.png)

In the DG, the `Component` class is said to be an abstract class but in the diagram, it isn't shown as an abstract class. If I didn't misunderstand the diagram, perhaps this is a mistake?

kairuler replied:
The class diagram is very small and hard to read.

sistine-yu replied:
The texts in the diagram is hard to read even if zooming in. Consider remove some unnecessary details or split the diagram into two sub diagrams to make it more legible.

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+Overall, this class diagram provides a high-level overview of the components that make up a computer system and their relationships to each other.
+
+### Command class
+#### `list [COMPONENT]` Command (builder mode)
```
Should the argument `[COMPONENT]` be included in the sub-section header? And does the (builder mode) need to be specified if there is no mention of another mode?

SpawnageLoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+found below:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![List Component Command](./uml-pictures/listComponentCommand.png)
```
> The BuilderListComponentCommand.executeCommand() method is then executed as part of the loop inside BigPP.runLoopUntilExit().

Should there be a loop in the diagram to represent this loop?

Magmanat replied:
not in scope

SpawnageLoong reviewed (commented):
Overall good explanation, but class diagrams tend to be excessively detailed to the point that they cannot be read without zooming in.

kairuler reviewed (commented):
Overall really nice DG :) I liked how some of the classes use a Initialization, Functionality, Behavior structure

notbingsu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+## Table of contents
+
+- [General Overview](#general-overview)
+- [Design \& implementation](#design--implementation)
```
`Design \& implementation`, could possibly have meant `Design & Implementation`

sistine-yu replied:
There's a typo and the `\` should be deleted.

notbingsu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+
+####  ViewerAddCommand() method
+This Sequential analysis will show how the ViewerAddCommand() method works, this will also serve as an example for the ViewerDeleteCommand and ViewerViewCommand.
+During the ParseCommand, when a add command is detected in the user input, it will call the `ViewerAddCommand` method with 
```
Type of variable can be defined, ie name could be specified of type string

notbingsu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
```
command may be deprecated

sistine-yu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+
+Overall, this class diagram provides a high-level overview of the components that make up a computer system and their relationships to each other.
+
+### Command class
```
Since the developer guide provides a comprehensive diagram of list Command, it is advised to add on other command methods analysis to be consistent. Very informative diagram.

sistine-yu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,201 @@
+## Design & implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### UI class
```
Maybe a multiplicity needs to be added between UI and UIState as well.

Magmanat committed:
Update UserGuide.md

Bugs with filter command

Magmanat committed:
Update UserGuide.md

Magmanat committed:
Update UserGuide.md

Commit:
Update UG

Commit:
Refactor table of contents

Magmanat committed:
Merge pull request #154 from typingpanda/user-guide

Update UG

Commit:
Add exception to account for interger overflow

Commit:
Add limit to integer size

Commit:
handle integer overflow

Commit:
Protect from integer overflow

Commit:
Fix grammer errors

Commit:
Fix checkstyle

Commit:
Magmanat

initial branch commit

Magmanat committed:
Add changes required to follow user guide

Reynold-SL committed:
Update ViewerFilterCommand to account for invalid inputs and empty commands
Update ViewerHelpCommand to include filter commands
Update return message in PCList

Reynold-SL committed:
Update checkstyle of code

Magmanat committed:
Solve clear screen bug for windows cmd terminal

Magmanat committed:
Add getDetails which will improve the list quality

Magmanat committed:
Ensure only selected sockets can be added

Magmanat committed:
Remove log file

Magmanat committed:
Solve bug with custom cpu and motherboard not adding because of socket

Magmanat committed:
Revert docs change

Magmanat committed:
Remove random ' from storage toString method

Magmanat committed:
Merge pull request #191 from typingpanda/fix-bugs

Fix bugs

Magmanat committed:
Merge pull request #193 from Reynold-SL/branch_viewer_filter_bugs

Fix ViewerFilterCommand bugs

Commit:
last changed at Apr 2, 2023 4:31 AM, pushed by Magmanat

Magmanat committed:
Fix Junit test to actual output

Magmanat committed:
Handle merge conflicts

Magmanat committed:
Disable shadowjar

Magmanat committed:
Enable shadowjar

Magmanat committed:
Edit EXPECTED.TXT to fit the new changes

keanecxz committed:
Merge pull request #197 from AY2223S2-CS2113-T12-2/user-guide-update

User guide update

Commit:
Add psu filter

Commit:
Fix checkstyle

Magmanat committed:
Rearrange tests

Reynold-SL committed:
Fix viewer filter to desplay accurate error message

Magmanat committed:
Remove commands from input.txt

Magmanat committed:
Merge pull request #198 from typingpanda/fix-bugs

Fix bugs

Magmanat committed:
Update README.md

keanecxz committed:
Merge pull request #199 from Reynold-SL/branch_viewer_2_filter_bugs

Fix viewer filter to diplay accurate error message

Magmanat committed:
Update UserGuide.md

keanecxz committed:
Merge pull request #192 from Magmanat/master

Add changes required to follow user guide

Magmanat committed:
Add appropriate nextlines

Magmanat committed:
Update UserGuide.md

Commit:
Update  User Guide Glossary Bug

Ryujikjs committed:
Merge branch 'master' into user-guide-update

Magmanat committed:
Merge pull request #200 from AY2223S2-CS2113-T12-2/user-guide-update

Update  User Guide Glossary Bug

Commit:
Update list component userguide

Commit:
Add spacing

Commit:
Add input formatting

Commit:
Add Prince PSU component

Commit:
Remove Prince of Pereira

Magmanat committed:
Merge pull request #201 from typingpanda/user-guide

Update list component userguide

Magmanat committed:
Update user guide with @@author

Magmanat committed:
Merge pull request #202 from Magmanat/master

Update user guide with @@author

Magmanat committed:
Edit author tags to appropriate format

superkaiba committed:
Add Thomas project portfolio

superkaiba committed:
Improve visibility of headings in DG

superkaiba committed:
Add color and indentation and bold/italics to PC viewer menu

Ryujikjs committed:
Reformat help commands to be more readable

Ryujikjs committed:
Fix checkstyle error

Magmanat committed:
Merge pull request #203 from superkaiba/add-thomas-project-portfolio

Add Thomas project portfolio

Magmanat committed:
Merge pull request #206 from Ryujikjs/format-help-command

Reformat help commands to be more readable

Magmanat committed:
Merge pull request #204 from superkaiba/improve-dg-formatting

Improve visibility of headings in DG

Commit:
Refactor guide to meet cs2113 standards

Magmanat committed:
Merge pull request #207 from typingpanda/developer-docs

Refactor guide to meet cs2113 standards

Magmanat committed:
Add to componentList and PCList methods

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp

Magmanat committed:
Update DataStorage class diagram

Magmanat committed:
Merge pull request #208 from Magmanat/master

Update DataStorage class diagram

Reynold-SL committed:
Update ViewerAddCommand to include the activation bar
Update ParserClass class diagram to follow standard practice for diagrams

Reynold-SL committed:
Merge branch 'master' into branch_error_developer_guide

# Conflicts:
#	docs/AboutUs.md
resolve merge conflicts

Ryujikjs committed:
Reformat PC class diagram and complete user guide and instructions for manual testing

Ryujikjs committed:
Resolve merge conflict

Ryujikjs committed:
Add profile picture

Magmanat committed:
Update ListComponentCommand PUML

Reynold-SL committed:
Update ViewerAddCommand to follow standard requirements

Commit:
Add object diagram to DG

keanecxz committed:
Merge pull request #209 from Reynold-SL/branch_error_developer_guide

Update developer guide

Ryujikjs committed:
Reformat PC class architecure to be more readable

Ryujikjs committed:
Resolve merge conflict

Ryujikjs committed:
Merge pull request #210 from Ryujikjs/developer-guide

Developer Guide Update

Magmanat committed:
Upgrade DataStorage sequential diagram

Magmanat committed:
Update loadChassisSequential diagram png

Magmanat committed:
Remove User from ListComponentCommand puml

Magmanat committed:
Update getIsFilter to getIsFiltered

Commit:
Fix merge conflict

Magmanat committed:
Rearrange Developer Guide properly

Commit:
changed explanation for abstract class of UI. Used connection instead of interface

Magmanat committed:
Ensure all Class in listComponentCommand has :

Magmanat committed:
Export new png listComponentCommand

Magmanat committed:
Settle merge conflicts

Magmanat committed:
Remove unecessary items from TOC

Magmanat committed:
Add line to differentiate items on DG

Ryujikjs committed:
Merge pull request #213 from Magmanat/master

Update PUML for loadChassisSequential and listComponentCommand

Commit:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp into developer-docs

Ryujikjs committed:
Merge pull request #212 from typingpanda/developer-docs

Add object diagram to DG

Magmanat committed:
Fix DataStorage Class Diagram to remove :

Magmanat committed:
Ensure PcDiagram correct sizing

Commit:
Fix correctness of diagrams in DG

Magmanat committed:
Add @@author to DG

Magmanat committed:
Add getting started and acknowledgements

superkaiba committed:
Fix PC Viewer menu tests

Ryujikjs committed:
Update Individual profile page, fix bug in user stories table and ammend PC class digram

Commit:
Add PPP

Magmanat committed:
Add fictional data for all components in resources

keanecxz committed:
Merge pull request #214 from Ryujikjs/PPP

Update PPP

Magmanat committed:
Solve merge conflicts

Commit:
Update PPP

Commit:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp into developer-docs

Ryujikjs committed:
Fix spelling errors

Magmanat committed:
Revert accidental change of PcDiagram

Magmanat committed:
Merge pull request #217 from Ryujikjs/PPP

Fix spelling errors

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp into Prince-PPP

Commit:
Fix Object diagram

Magmanat committed:
Merge pull request #215 from typingpanda/developer-docs

Add PPP

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp into Prince-PPP

Magmanat committed:
Settle merge conflicts and add items to PPP

Magmanat committed:
Add @@author for appendix

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp

Ryujikjs committed:
Merge pull request #216 from Magmanat/master

Changes to improve the DG

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp into Prince-PPP

Magmanat committed:
Add Prince PPP to ensure all the details are added

Magmanat committed:
change aboutUs to reflect correct portfolio

superkaiba committed:
Fix more PC Viewer menu tests

superkaiba committed:
remove text-ui-test

superkaiba committed:
Remove dos2unix

Magmanat committed:
Add Reviewing/Mentoring Contributions

Magmanat committed:
Add team contributions section

Reynold-SL committed:
Update ViewerAddCommand in the developer guide.
Change ParseViewerCommand to Parser
Added activation bar for Parser
Update argument description to be of type String

superkaiba committed:
Merge pull request #219 from Reynold-SL/branch_viewer_developer_guide

Update ViewerAddCommand in the developer guide.

superkaiba committed:
Merge pull request #218 from Magmanat/Prince-PPP

Addition of Prince PPP to the Team portfolios

Magmanat committed:
Resolve merge conflicts

Magmanat committed:
Change to lowercase for PPP

Magmanat committed:
Merge pull request #220 from Magmanat/master

Minor changes to Prince PPP

superkaiba committed:
Remove color from terminal

Magmanat committed:
Remove conflict marker

Magmanat committed:
Change to lowercase

Magmanat committed:
Merge pull request #205 from superkaiba/format-pc-viewer-menu-better

Add color and indentation and bold/italics to PC viewer menu

superkaiba committed:
Modify Thomas PPP

Commit:
Update userguide for budget command

Commit:
Add documentation for componentlistcommand

Commit:
Modify PPP

Reynold-SL committed:
Update portfolio

Reynold-SL committed:
Update portfolio to add team based tasks
Update about us with picture
Update dg to reflect accurate github name

superkaiba committed:
Merge pull request #224 from Reynold-SL/branch_portfolio

Update portfolio

superkaiba committed:
Merge pull request #223 from typingpanda/documentation

Add documentation for componentlistcommand

superkaiba committed:
Merge pull request #222 from typingpanda/user-guide

Update userguide for budget command

Magmanat committed:
Merge pull request #221 from superkaiba/modify-thomas-ppp

Modify Thomas PPP

Magmanat committed:
Update AboutUs.md

Ryujikjs committed:
Add Junit test for compare commands and change PPP file naming

Ryujikjs committed:
Fix spelling error

Magmanat committed:
Merge pull request #225 from Ryujikjs/More-Junit

Add Junit test for compare commands

Magmanat committed:
Rename Ryujikjs.md to ryujikjs.md

Magmanat committed:
Update AboutUs.md

Magmanat committed:
Rename KeaneChan.md to typingpanda.md

Magmanat committed:
Update AboutUs.md

Magmanat committed:
Update magmanat.md

Reynold-SL committed:
Create JUnit Test

Reynold-SL committed:
Update checkstyle of code

Reynold-SL committed:
Update ViewCommandTest according to checkstyle

Reynold-SL committed:
Fix errors with FilterCommandTest and EditCommandTest

Reynold-SL committed:
Edit Developer Guide to include outcomes for each example
Update ViewerAddCommand sequence diagram and description
Update UserGuide on the notes for the filter command and output for the help command

Magmanat committed:
Merge pull request #226 from Reynold-SL/branch_junit_test

Create JUnit Test

Magmanat committed:
Update to follow format of sequential diagram for Commands

Magmanat committed:
Merge pull request #228 from Magmanat/update_reynold_pr

Update reynold pr

Magmanat committed:
Update AboutUs.md

Magmanat committed:
Update UserGuide.md

Magmanat committed:
Add Page breaks to UG

Ryujikjs committed:
Update PPP to have page break at top and add additional information

keanecxz committed:
Merge pull request #229 from Ryujikjs/PPP-Update

Update PPP to have page break at top and add additional information

Commit:
Update PPP

Commit:
Fix merge conflict

Commit:
Update PPP

Commit:
Fix spelling error

keanecxz committed:
Merge pull request #230 from typingpanda/PPP

Update PPP

keanecxz committed:
Remove page break for keane PPP

Ryujikjs committed:
Remove line break

Ryujikjs committed:
Merge pull request #231 from Ryujikjs/PPP-Update

Remove line break

Reynold-SL committed:
Update Reynold's PPP

Magmanat committed:
Update UserGuide.md

Reynold-SL committed:
Fix Formatting issue for Reynold's PPP

Magmanat committed:
Update DeveloperGuide.md

Magmanat committed:
Update DeveloperGuide.md

Magmanat committed:
Update DeveloperGuide.md

Magmanat committed:
Update DeveloperGuide.md

Ryujikjs committed:
Rynold PPP  update

Reynold-SL committed:
Merge remote-tracking branch 'origin/master'

Reynold-SL committed:
Merge pull request #232 from Ryujikjs/PPP-Update

Rynold PPP  update

Magmanat committed:
Update DeveloperGuide.md

Magmanat committed:
Update magmanat.md

Magmanat committed:
Update magmanat.md

Magmanat committed:
Update magmanat.md

Magmanat committed:
Update magmanat.md

Magmanat committed:
Update magmanat.md

Magmanat committed:
Add Exception handling for DataStorage

Magmanat committed:
Update for DataStorage loading exception

Magmanat committed:
New checks to ensure tampering with data file is handled properly

Magmanat committed:
Merge pull request #233 from Magmanat/master

New checks to ensure tampering with data file is handled properly

Magmanat committed:
Update UserGuide.md

Magmanat committed:
Update UserGuide.md

Magmanat committed:
Ensure efficiency can only be gold, silver, bronze

Magmanat committed:
Update UserGuide.md

Reynold-SL committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/team/reynold-sl.md
resolve conflicts

Magmanat committed:
Ensure custom commands line up with UG

Reynold-SL committed:
New Updates to Reynold's PPP

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp

Magmanat committed:
Merge pull request #234 from Reynold-SL/branch_error_Reynold_PPP

Branch error reynold ppp

Magmanat committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-2/tp

Magmanat committed:
Change PC VIEWER to PC viewer

Magmanat committed:
Update UserGuide.md

Magmanat committed:
Update DeveloperGuide.md

Magmanat committed:
Change the space divider for split

Magmanat committed:
Solve some bugs on IO

Magmanat committed:
Merge pull request #235 from Magmanat/master

Minor bug fixes

Magmanat committed:
Ensure random input after valid delete, view, edit is handled

Magmanat committed:
Change how custom command handles price

Magmanat committed:
Add funny message for too high of a price

Commit:
Fix exception messages

Magmanat committed:
Remove float price from custom gpu accidentally

Magmanat committed:
Merge pull request #236 from Magmanat/master

Minor changes to price handling

Commit:
Fix test exception messsage

Commit:
Fix ram json memory correctness

Commit:
Fix list ram memory flag

Commit:
Fix ram exception message

Commit:
Fix ram excecption message again

Magmanat committed:
Handle ParseFloat infinity

Magmanat committed:
Merge pull request #237 from typingpanda/fix-bugs

Fix bugs for exception message in listcommand

Magmanat committed:
Change some formatting errors

Reynold-SL committed:
Fix userguide and Dg

Reynold-SL committed:
Merge pull request #239 from Reynold-SL/branch_update_user_guide

Fix userguide and Dg

Magmanat committed:
Change regex expression

Magmanat committed:
Merge pull request #238 from Magmanat/master

Handle ParseFloat infinity

Commit:
Fix out of bounds error

Commit:
Fix naming error

Commit:
Fix float parsing

Commit:
Fix Junit test

Commit:
Fix merge conflict

Commit:
Fix parsing for integers and floats

Commit:
Fix ranges

Commit:
Fix ranges

Ryujikjs committed:
Update user guide flags

Commit:
Fix errors

Ryujikjs committed:
Update user guide flags

Magmanat committed:
Merge pull request #240 from typingpanda/fix-bugs

Fix bugs
